aws/resource_aws_storagegateway_gateway.go

```diff
@@ -391,6 +395,19 @@ func resourceAwsStorageGatewayGatewayCreate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	if v, ok := d.GetOk("smb_file_share_visibility"); ok {
+		input := &storagegateway.UpdateSMBFileShareVisibilityInput{
+			GatewayARN:        aws.String(d.Id()),
+			FileSharesVisible: aws.Bool(v.(bool)),
+		}
+
+		log.Printf("[DEBUG] Storage Gateway Gateway %q setting SMB File Share Visibility", input)
+		_, err := conn.UpdateSMBFileShareVisibility(input)
+		if err != nil {
+			return fmt.Errorf("error setting SMB File Share Visibility: %w", err)
```

Comment:
I see that most of the current error messages for this resource don't include the ID but let's take this opportunity for improvement.

```suggestion
			return fmt.Errorf("error updating Storage Gateway Gateway (%s) SMB file share visibility: %w", d.Id(), err)
```